Microsoft.DependencyInjection.SolrNet/ServiceCollectionExtensions.cs

```diff
@@ -86,5 +88,19 @@ private static IServiceCollection AddSolrNet(this IServiceCollection services, I
 
             return services;
         }
+
+        /// <summary>
+        /// Method to deal with adding a second core into Microsoft's dependency injection system.
+        /// </summary>
+        /// <param name="services">The dependency injection service.</param>
+        /// <param name="url">The url for the second core.</param>
+        /// <typeparam name="TModel">The type of model that should be used for this core.</typeparam>
+        /// <returns>The dependency injection service.</returns>
+        public static IServiceCollection AddSolrCore<TModel>(this IServiceCollection services, string url)
```

Comment:
I'm a little bit on the fence regarding the naming. 
It's a little confusing that:
1) I first have to call `AddSolrNet` (and already pass in a URL, but no type)
2) Then for the next core call `AddSolrCore<> ()` and pass in another URL with a type.

Perhaps we could go the same way as DbContext:
``` csharp
services.AddDbContext<SomeContext>(options => someoptions) ;
services.AddDbContext<OtherContext>(options => someotheroptions);
```

So in our case:
Legacy:
`public static IServiceCollection AddSolrNet(this IServiceCollection services, string url)`
New:
` public static IServiceCollection AddSolrNet<TModel>(this IServiceCollection services, string url)`

I would like to see that calling the new method, will also work without the need to call the legacy one. 
So we should take the 'general DI setup' out of the legacy method, and into a private helper, and make sure inside that helper method that it does its work only once.

So the goal is that in my `startup.cs` I would be able to write:

services.AddSolrNet<Foobar>("http://solr.com/foobar");
services.AddSolrNet<Testing>("http://solr.com/testing");